test/errors_test.cpp

```diff
@@ -485,7 +485,13 @@ INSTANTIATE_TEST_CASE_P(StatementsTest_2, ErrorsGenericTest, ::testing::Values(
                             InputOutputPair{"{% if a == 42 %}{% endwith %}",
                                             "noname.j2tpl:1:20: error: Unexpected statement: 'endwith'\n{% if a == 42 %}{% endwith %}\n                ---^-------"},
                             InputOutputPair{"{{}}",
-                                            "noname.j2tpl:1:3: error: Unexpected token: '<<End of block>>'\n{{}}\n--^-------"}
+                                            "noname.j2tpl:1:3: error: Unexpected token: '<<End of block>>'\n{{}}\n--^-------"},
+                            InputOutputPair{"{% raw %}{% raw %}{{ x }{% endraw %}{% endraw %}",
+                                            "noname.j2tpl:1:37: error: Unexpected raw block end\n{% raw %}{% raw %}{{ x }{% endraw %}{% endraw %}\n                                 ---^-------"},
+// FIXME:                    InputOutputPair{"{% raw %}",
```

Comment:
Still not sure how to handle this case?

How should I solve this? @flexferrum

Reply:
I suppose the first `endraw` should close the first `raw` block and the second `endraw` should be unexpected.

Reply:
Yes, it is ok. I saw that in this page https://jbmoelker.github.io/jinja-compat-tests/tags/raw/

That case I think is working properly.

The case that I am not sure how to handle is this one:

`{% raw %}` I mean, the error when the regex_iterator arrives to the EOF and the RawBlock has not been closed.

Reply:
I think in this case it's necessary to return `ExpectedRawEnd` error but points to the end of the `{% raw %}` tag.

Reply:
Yes, I mean, I have to modify the test InputOutputPair values, but the main error here is that this error is not being thrown because I am not sure where I should return `ExpectedRawEnd`.

Maybe you could suggest me where I should look at.

Reply:
I see. I suppose, you can do it at this line: https://github.com/jinja2cpp/Jinja2Cpp/blob/bc8c2c270eabd69e8f93c5a43df29b77c03586df/src/template_parser.h#L355

At this point you can check that the current block is finished or it is raw text. Otherwise you can return an error depends on the unclosed block type.